test/streams-test.js

```diff
@@ -337,3 +331,46 @@ describe("Streaming sugar", function() {
     });
 });
 
+describe("Encoding using internal modules: with surrogates in separate chunks", function () {
+    it("normal", checkEncodeStream({
+        encoding: "utf8",
+        input: ["\uD83D", "\uDE3B"],
+        output: "f09f98bb",
+    }));
+
+    it("more than one character with left", checkEncodeStream({
+        encoding: "utf8",
+        input: ["abc\uD83D", "\uDE3B"],
+        output: "616263f09f98bb",
+    }));
+
+    it("more than one character with right", checkEncodeStream({
+        encoding: "utf8",
+        input: ["\uD83D", "\uDE3Befg"],
+        output: "f09f98bb656667",
+    }));
+
+    it("more than one character at both ends", checkEncodeStream({
+        encoding: "utf8",
+        input: ["abc\uD83D", "\uDE3Befg"],
+        output: "616263f09f98bb656667",
+    }));
+
+    it("surrogates pair be interrupted", checkEncodeStream({
+        encoding: "utf8",
+        input: ["abc\uD83D", "efg\uDE3B"],
+        output: "616263656667",
```

Comment:
The output we expect should be equivalent to what Node does on a merged input chunks:
```
> new Buffer("abc\uD83D" + "efg\uDE3B", "utf8").toString("hex")
'616263efbfbd656667efbfbd'
```